### DISCUSSION TITLE
What happens if I have records from different languages?

### DISCUSSION THREAD

**Discussion by SagevdBrand** [asreview/asreview#551] [Q&A] [answered]
Say, I have a dataset where there are also articles in Dutch and German, aside from the most being in English, does ASReview then still function properly?

**Comment by terrymyc** [comment, marked as answer]
The texts of the records are handled as is. ASReview does not differentiate between records that use different languages. Therefore, ASReview will have difficulty with identifying a relevant record when it is written in a language that is different from the rest of the records in your dataset.

**Reply by brittdev** [reply]
Do you perhaps know what the best way is to deal with having records in different languages then? I was thinking that maybe you could keep the articles in different languages separate from each other and upload them separately in ASReview, but I wonder if that doesn't take away the essence of time saving (since shorter longlists save less time). Or would it help if you include all the different languages in the training data? I also found some literature about methods/models (Bert, XLM-R) that enable multilingual document classification but I'm not skilled enough in Python to write a script and implement it in ASReview.

**Reply by jteijema** [reply]
I am currently working on a language invariant feature extractor (see https://huggingface.co/bert-base-multilingual-cased). 

Say you have the same abstract in german and in english, then this feature extractor would assign the same values to these abstracts for classification, thereby removing the language barrier.

It will be a short while before this will be publically available, but I can share it in the form of an extension, if you are interested, as soon as the first development phases are ready to use.

**Reply by brittdev** [reply]
Thanks for your quick reply! That sounds very promising and like something that would be very useful in my research. Can I contact you about it via email?

**Reply by jteijema** [reply]
Yes, ofcourse! You can reach me at j.j.teijema@uu.nl.

**Reply by jteijema** [reply]
https://github.com/asreview/asreview-multilingual-feature-extractor is ready to use! We're still testing, so feedback is very welcome. It should work with the following languages:
Arabic, Chinese, Dutch, English, French, German, Italian, Korean, Polish, Portuguese, Russian, Spanish, Turkish.

**Reply by brittdev** [reply]
My apologies for getting back to you this late @JTeijema, but thank you very much for working on this! It turns out that most datasets I'll be using in my research don't have this issue, but I will try the extension out on one that does contain records in multiple languages and let you know how this affects my results!

**Reply by samhefnawy** [reply]
Is Arabic well enabled now? how can we test it?
Has anyone used it for Arabic Texts - can you share that and connect me with that who did, if possible?

**Comment by Emanuel-1986** [comment]
@jteijema 

I am planning to use ASReview for my scoping review. Has there been any development regarding screening articles in languages other than English? 

Thank you. 
Emanuel

**Reply by jteijema** [reply]
Yes! We've been testing extensively using the new multilingual models (https://github.com/Robdboer/Multilingual-Feature-Extractors-Study). We're working on new model extensions for asreview, one of those will be the multilingual feature extractor (https://jteijema.github.io/synergy-simulations-website/models.html#labse, https://jteijema.github.io/synergy-simulations-website/models.html#multilingual). 

We're currently in the process of publishing a paper this topic. But if you want to try my personal experiments, visit [my experimental model repo](https://github.com/jteijema/ASReview-experimental-models), and try the labse model.

**Reply by Emanuel-1986** [reply]
Dear @jteijema thank you for your response.

**Reply by JonnaLind** [reply]
Dear @jteijema, 
Colleagues of mine (I'm in the library) are planning a review of Dutch newspaper articles. Is the current version of ASReview equipped to deal with Dutch language input?
Groetjes, Jonna (ARQ Library)

**Reply by jteijema** [reply]
Hi @JonnaLind ,

If you're using a single language, then try out TF-IDF. It's optimized with stopword removal for English but I reckon Dutch will work. We're lacking a Dutch dataset to run proper simulations on. If you happen to have one, I can give you a definitive answer, hahaha.

Another option is to use the LaBSE model I've mentioned above, by installing https://github.com/jteijema/ASReview-experimental-models. We'll have the new model package repository available soon that will simplify the installation of the package.
